packages/next/src/server/use-cache/cache-life.ts

```
@@ -0,0 +1,141 @@
+    }
+  }
+
+  if (profile.revalidate !== undefined && profile.expire !== undefined) {
+    if (profile.revalidate > profile.expire) {
+      throw new Error(
+        'If providing both the revalidate and expire options, ' +
+          'the expire option must be greater than the revalidate option.' +
+          'The expire option indicates how many seconds from the start ' +
+          'until it can no longer be used.'
+      )
+    }
+  }
+
+  if (profile.stale !== undefined && profile.expire !== undefined) {
+    if (profile.stale > profile.expire) {
+      throw new Error(
+        'If providing both the stale and expire options, ' +
+          'the expire option must be greater than the stale option.' +
+          'The expire option indicates how many seconds from the start ' +
+          'until it can no longer be used.'
+      )
+    }
+  }
+}
+
+export function cacheLife(profile: CacheLifeProfiles | CacheLife): void {
+  if (!process.env.__NEXT_DYNAMIC_IO) {
+    throw new Error(
+      'cacheTag() is only available with the experimental.dynamicIO config.'
+    )
+  }
+
+  const workUnitStore = workUnitAsyncStorage.getStore()
+  if (!workUnitStore || workUnitStore.type !== 'cache') {
+    throw new Error(
+      'cacheLife() can only be called inside a "use cache" function.'
+    )
+  }
+
+  if (typeof profile === 'string') {
+    const configuredProfile = cacheLifeProfileMap.get(profile)
+    if (configuredProfile === undefined) {
+      if (cacheLifeProfileMap.has(profile.trim())) {
+        throw new Error(
+          `Unknown cacheLife profile "${profile}" is not configured in next.config.js\n` +
+            `Did you mean "${profile}" without the spaces?`
+        )
+      }
+      throw new Error(
+        `Unknown cacheLife profile "${profile}" is not configured in next.config.js\n` +
+          'module.exports = {\n' +
+          '  cacheLife: {' +
+          '    "...": ...' +
+          '  }' +
+          '}'
```

for the future we should expand the ts plugin to generate this type so we can fail if you pass an invalid string statically